TESTING.md

```diff
@@ -108,7 +107,18 @@ The test name is `test_` concatenated with the name of the test listed in the te
 You can get a local coverage report with
 `COVERAGE=1 bundle exec ruby -Itest ./path/to/your/test.rb`
 
-If you get an error about missing db fields, try migrating your test database:
+#### Common issues and potential fixes
+
+1. If you get a bunch of complaints about database, like missing tables or how some tables haven't been seeded, here are some things you can try in order from least to most drastic before running your tests again:
+
+    1. Stop spring process (which can sometimes have stale data) and then rerun the tests, which will automatically start a new instance of spring.
+      `spring stop` 
```

Comment:
Good call out - I noticed there were the below processes running when I checked. Running spring stop said, `Spring is not running`

Do you know what these are and what would be the guidance in case someone hits this situation. 

Another related question - does a spring process run when the dashboard server is run?

vijaya@VijiMiniDskTp:~/source/code-dot-org$ ps -ef | grep spring
vijaya    590106  525085  0 09:20 pts/0    00:00:00 spring server | dashboard | started 18 hours ago
vijaya    639978  590106 35 12:01 ?        01:53:55 spring app    | dashboard | started 5 hours ago | test mode
vijaya    733855  525086  0 17:19 pts/0    00:00:00 grep --color=auto spring

Reply:
Strangely, when I ran the command while running a test, it still gave the same output, i.e., there were no new processes added to that list.

Reply:
> Another related question - does a spring process run when the dashboard server is run?

Good question, it does not for me because I have this environment variable set:
```
DISABLE_SPRING=true
```

my recollection is that spring does run with bin/dashboard-server when that environment variable is not set, but it is possible that has changed.

Reply:
I'm thinking I'll skip this update given its not conclusive how we can determine if spring stopped. But, I've had good success with running spring stop so far when hitting issues after DB migrate. Let me know what you think.

Reply:
yes, good points. your additions will be a nice improvement to the docs, and we can always add more later if people need more guidance on stopping spring.